srtp-ekt-diet/draft-ietf-perc-srtp-ekt-diet.md

```diff
@@ -599,171 +599,201 @@ frames. If only sending audio, the RECOMMENDED frequency is every
 management. DTLS-SRTP can be viewed in two equivalent ways: as a new
-key management method for SRTP, and a new RTP-specific data format for
-DTLS.
-  
+key management method for SRTP, and a new RTP-specific data format
+for DTLS.
+
 
-## SRTP EKT Key Transport Extensions to DTLS-SRTP {#dtls-srtp-extensions}
-      
-This document defines a new TLS negotiated extension called
-"srtp\_ekt\_key\_transport" and a new TLS content type called
-EKTMessage.
+## SRTP EKT Key Transport Extensions to DTLS-SRTP
+{#dtls-srtp-extensions}
 
-Using the syntax described in DTLS [@!RFC6347], the following
-structures are used:
+This document defines a new TLS negotiated extension
+`supported_ekt_ciphers` and a new TLS handshake message type
+`ekt_key`.  The extension negotiates the cipher to be used in
+encrypting and decrypting EKTCiphertext values, and the handshake
+message carries the corresponding key.
+
+The diagram below shows a message flow of DTLS 1.3 client and server
+using EKT configured using the DTLS extensions described in this
+section.  (The initial cookie exchange is omitted.)
 
-{#tls-datastructure}
 ~~~
-enum {
-  reserved(0),  
-  aeskw_128(1),  
-  aeskw_256(2),  
-} EKTCipherType;  
+Client                                             Server
 
-struct {
-  EKTCipherType ekt_ciphers<1..255>;
-} SupportedEKTCiphers;
+ClientHello
+ + key_share*
+ + pre_shared_key*      -------->
+ + cookie
+ + use_srtp
+ + supported_ekt_ciphers
+                                               ServerHello
+                                              + key_share*
+                                         + pre_shared_key*
+                                                + use_srtp
+                                   + supported_ekt_ciphers
+                                     {EncryptedExtensions}
+                                     {CertificateRequest*}
+                                            {Certificate*}
+                                      {CertificateVerify*}
+                        <--------               {Finished}
 
-struct {
-  EKTCipherType ekt_cipher; 
-  uint ekt_key_value<1..256>;  
-  uint srtp_master_salt<1..256>;  
-  uint16 ekt_spi;  
-  uint24 ekt_ttl; 
-} EKTkey;  
 
+{Certificate*}
+{CertificateVerify*}
+{Finished}              -------->
+
+                        <--------                    [Ack]
+
+                        <--------                 [EKTKey]
+[Ack]                   -------->
+
+|SRTP packets|          <------->           |SRTP packets|
++ <EKT tags>                                  + <EKT tags>
+
+
+{} Messages protected using DTLS handshake keys
+
+[] Messages protected using DTLS application traffic keys
+
+<> Messages protected using the EKTKey and EKT cipher
+
+|| Messages protected using the SRTP Master Key sent in
+   a Full EKT Tag
+~~~
+
+When used in multi-party SRTP sessions, the DTLS Server may be a different
+entity than the source of the SRTP packets.  If the DTLS-SRTP server performs a
```

Comment:
Isn't the source fof the SRTP packeta always the dtls client (so not the server)

Reply:
No.  In a 1-1 WebRTC call, for example, both the client and the server emit packets.  The point is that here the server might only serve a key synchronization function, as in PERC.  I agree the wording is unclear; will revise.